src/app/api/verify/route.ts

```diff
@@ -0,0 +1,52 @@
+import { createJob, createTransactions } from '@/actions/job.action';
+import { NextRequest, NextResponse } from 'next/server';
+import { validateWebhookSignature } from 'razorpay/dist/utils/razorpay-utils';
+
+export async function POST(req: NextRequest) {
+  const jsonBody = await req.json();
+  const razorpay_signature: string | null = req.headers.get(
+    'X-Razorpay-Signature'
+  );
+
+  if (!razorpay_signature)
+    return NextResponse.json({ error: 'Signature not found' }, { status: 404 });
+
+  const isPaymentValid: boolean = validateWebhookSignature(
+    JSON.stringify(jsonBody),
+    razorpay_signature,
+    process.env.RAZORPAY_WEBHOOK_SECRET!
+  );
+
+  if (!isPaymentValid) {
+    return NextResponse.json(
+      { error: 'Payment not verified. Payment signature invalid' },
+      { status: 404 }
+    );
+  }
+
+  const jobData = jsonBody.payload.payment.entity.notes;
```

Comment:
we have lodash library in deps , lets leverage the _.get function to access nested values